kernel/src/platform/mod.rs

```diff
@@ -13,6 +13,13 @@ pub trait Platform {
     fn with_driver<F, R>(&self, driver_num: usize, f: F) -> R
     where
         F: FnOnce(Option<&Driver>) -> R;
+
+    fn with_driver_permissions<F, R>(&self, _permissions: u64, driver_num: usize, f: F) -> R
```

Comment:
add a comment explaining what this method does, and that it's default implementation is a no-op (i.e. it just blindly calls `with_driver`)